salesforce/salesforce-inbound-endpoint/salesforce-inbound-endpoint-1.0.0/src/main/java/org/wso2/carbon/inbound/salesforce/poll/SalesforceStreamData.java

```diff
@@ -0,0 +1,342 @@
+            options.put(ClientTransport.TIMEOUT_OPTION, readTimeout);
+            LongPollingTransport transport = new LongPollingTransport(options, httpClient) {
+
+                @Override
+                protected void customize(ContentExchange exchange) {
+                    super.customize(exchange);
+                    exchange.addRequestHeader("Authorization", "OAuth " + sessionId);
+                }
+            };
+
+            BayeuxClient client = new BayeuxClient(salesforceStreamingEndpoint(endpoint), transport);
+
+            if (useCookies) establishCookies(client, userName, sessionId);
+            return client;
+        } catch (MalformedURLException e) {
+            handleException("Error while building URL: " + e.getMessage(), e);
+        } catch (Exception e) {
+            handleException("Error while starting the Http Client" + e.getMessage(), e);
+        }
+        return null;
+    }
+
+    private String salesforceStreamingEndpoint(String endpoint) throws MalformedURLException {
+        return new URL(endpoint + streamingEndpointUri).toExternalForm();
+    }
+
+    /**
+     * establish the Cookies fot the http client.
+     *
+     * @param client The Bayeux Client that connect with Salesforce Streaming Server.
+     * @param user   The User Name of Salesforce.
+     * @param sid    The session Id of Salesforce.
+     */
+    private void establishCookies(BayeuxClient client, String user, String sid) {
+        client.setCookie(SalesforceConstant.COOKIE_LOCALEINFO_KEY, SalesforceConstant.COOKIE_LOCALEINFO_DEFAULT_VALUE, waitTime);
+        client.setCookie(SalesforceConstant.COOKIE_LOGIN_KEY, user, waitTime);
+        client.setCookie(SalesforceConstant.COOKIE_SESSION_ID_KEY, sid, waitTime);
+        client.setCookie(SalesforceConstant.COOKIE_LANGUAGE_KEY, SalesforceConstant.COOKIE_LANGUAGE_DEFAULT_VALUE, waitTime);
+    }
+
+    /**
+     * load essential property for salesforce inbound endpoint.
+     *
+     * @param properties The mandatory parameters of Salesforce.
+     */
+    private void loadMandatoryParameters(Properties properties) {
+        if (log.isDebugEnabled()) {
+            log.debug("Starting to load the salesforce credentials");
+        }
+
+        userName = properties.getProperty(SalesforceConstant.USER_NAME);
+        salesforceObject = properties.getProperty(SalesforceConstant.SOBJECT);
+        password = properties.getProperty(SalesforceConstant.PASSWORD);
+        loginEndpoint = properties.getProperty(SalesforceConstant.LOGIN_ENDPOINT);
+        packageName = properties.getProperty(SalesforceConstant.PACKAGE_NAME);
+        packageVersion = properties.getProperty(SalesforceConstant.PACKAGE_VERSION);
+
+        if (StringUtils.isEmpty(userName) && StringUtils.isEmpty(salesforceObject) &&
+                StringUtils.isEmpty(password) && StringUtils.isEmpty(loginEndpoint) &&
+                StringUtils.isEmpty(packageName) && StringUtils.isEmpty(packageVersion)) {
```

Comment:
Fixed in https://github.com/wso2-attic/esb-connectors/pull/557/commits/c9ffbe2fa8d7cd7e3c7517aba0ad656a291a1350